packages/ngtools/webpack/src/transformers/import_factory.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+
+
+/**
+ * Given this original source code:
+ *
+ * import { NgModule } from '@angular/core';
+ * import { Routes, RouterModule } from '@angular/router';
+ *
+ * const routes: Routes = [{
+ *   path: 'lazy',
+ *   loadChildren: () => import('./lazy/lazy.module').then(m => m.LazyModule).
+ * }];
+ *
+ * @NgModule({
+ *   imports: [RouterModule.forRoot(routes)],
+ *   exports: [RouterModule]
+ * })
+ * export class AppRoutingModule { }
+ *
+ * NGC (View Engine) will process it into:
+ *
+ * import { Routes } from '@angular/router';
+ * const ɵ0 = () => import('./lazy/lazy.module').then(m => m.LazyModule);
+ * const routes: Routes = [{
+ *         path: 'lazy',
+ *         loadChildren: ɵ0
+ *     }];
+ * export class AppRoutingModule {
+ * }
+ * export { ɵ0 };
+ *
+ * The importFactory transformation will only see the AST after it is process by NGC.
+ * You can confirm this with the code below:
+ *
+ * const res = ts.createPrinter().printNode(ts.EmitHint.Unspecified, sourceFile, sourceFile);
+ * console.log(`### Original source: \n${sourceFile.text}\n###`);
+ * console.log(`### Current source: \n${currentText}\n###`);
+ *
+ * At this point it doesn't yet matter what the target (ES5/ES2015/etc) is, so the original
+ * constructs, like `class` and arrow functions, still remain.
+ *
+ */
+
+export function importFactory(
+  warningCb: (warning: string) => void,
+): ts.TransformerFactory<ts.SourceFile> {
+  return (context: ts.TransformationContext) => {
+    // Only transform files that contain `loadChildren`.
+    return (sourceFile: ts.SourceFile) => {
+      if (/\bloadChildren\b/.test(sourceFile.text)) {
+        const warning = `
+Found 'loadChildren' with a non-string syntax in ${sourceFile.fileName} but could not transform it.
+Make sure it matches the format below:
+
+loadChildren: () => import('IMPORT_STRING').then(m => m.EXPORT_NAME)
+
+Please note that only IMPORT_STRING and EXPORT_NAME can be replaced in this format.`;
+        const emitWarning = () => warningCb(warning);
+        const visitNode: ts.Visitor = (node: ts.Node) => {
+          // Only try to transform 'loadChildren' property assignments.
+          if (ts.isVariableDeclaration(node)) {
```

Comment:
Does ngc generate these for anything else?  Should we verify that the declaration is assigned to `loadChildren`?

Reply:
AFAIK ngc generates these `ɵ0 = something` nodes when removing functions from decorators so they can be exported to factories. Which means the current warning would be emitted for other things that we don't want. 

To help prevent this I moved the warning to only show when `ɵ0 = () => import('IMPORT_STRING').then(something)` is found onwards.

Before I figured out this transform only got the NGC transformed code I actually verified that it was on a `loadChildren`. But then came the indirection of `const ɵ0` which made that harder.

I tried to get from the variable declaration to its references with the type checker and couldn't find a good way. I could save the reference and then further visit all other nodes (like we do in some other transforms) but that would not be very performant.

 Since we were already aiming for support on a very specific use case (switching back between Ivy and VE and with a experimental flag) I thought it wouldn't be worth the work to make it much more robust.

I think that warning later in the structure reduces false negatives (warning about something that isn't a loadChildren import). But I do agree false positives (altering something that isn't a loadChildren import) are a problem. Do you have any suggestion on how to verify this in an efficient manner?